pkg/providers/vsphere/vmlifecycle/update_status.go

```diff
@@ -432,6 +436,43 @@ func MarkBootstrapCondition(
 	}
 }
 
+func MarkVMClassResizeSynced(
+	ctx context.Context,
+	vm *vmopv1.VirtualMachine,
+	k8sClient ctrlclient.Client) {
+
+	className := vm.Spec.ClassName
+	if className == "" {
+		conditions.Delete(vm, vmopv1.VirtualMachineResizeClassSynced)
+		return
+	}
+
+	if _, _, _, ok := vmopv1util.GetLastResizedAnnotation(*vm); !ok {
+		conditions.MarkUnknown(vm, vmopv1.VirtualMachineResizeClassSynced, "NoLastResizedAnnotation", "")
+		return
+	}
+
+	// Depending on what we did in the prior session update code for this VM, we might already
+	// have fetched the class but the way the code is structured today, it isn't very easy for
+	// us to pass that to here. So just refetch it again.
+	vmClass := vmopv1.VirtualMachineClass{}
+	err := k8sClient.Get(ctx, ctrlclient.ObjectKey{Name: className, Namespace: vm.Namespace}, &vmClass)
+	if err != nil {
+		if ctrlclient.IgnoreNotFound(err) == nil {
```

Comment:
For easier readability, suggest using `if apierrorsutil.IsNotFound(err) {`

Reply:
I was kind of aiming for one less import. The currently imported `apierrorsutil` isn't the package that has that method.